podman_compose.py

```diff
@@ -2784,28 +2796,25 @@ async def compose_up(compose: PodmanCompose, args):
         .splitlines()
     )
     diff_hashes = [i for i in hashes if i and i != compose.yaml_hash]
-    if args.force_recreate or len(diff_hashes):
+    if (args.force_recreate and len(hashes) > 0) or len(diff_hashes):
         log.info("recreating: ...")
         down_args = argparse.Namespace(**dict(args.__dict__, volumes=False))
         await compose.commands["down"](compose, down_args)
         log.info("recreating: done\n\n")
     # args.no_recreate disables check for changes (which is not implemented)
 
-    podman_command = "run" if args.detach and not args.no_start else "create"
-
     await create_pods(compose, args)
     for cnt in compose.containers:
         if cnt["_service"] in excluded:
             log.debug("** skipping: %s", cnt["name"])
             continue
-        podman_args = await container_to_args(
-            compose, cnt, detached=args.detach, no_deps=args.no_deps
-        )
-        subproc = await compose.podman.run([], podman_command, podman_args)
-        if podman_command == "run" and subproc is not None:
+        podman_args = await container_to_args(compose, cnt, detached=False, no_deps=args.no_deps)
+        subproc = await compose.podman.run([], "create", podman_args)
+        if not args.no_start and args.detach and subproc is not None:
```

Comment:
This should be extracted to a separate PR, because if something breaks I will have easier time bisecting.

Reply:
Well, this block of change is crutial to the issue #1183. It is vital to depends_on condition check by using `create` rather than `run` to make a container in the first pass. If we `run` the container in the first place, the depends_on check is bypassed. This change also fixes the double-execution problem (#1176) in detached mode. So this change just happens to resolve multiple issues.  I hope this addresses your concern.